mobile/src/navigation/app/components/drawer-content/drawer-content.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { FC } from 'react';
+import { ScrollView, Button, Image } from 'react-native';
+import {
+  DrawerContentComponentProps,
+  DrawerItem,
+} from '@react-navigation/drawer';
+import { View, Text } from '~/components/common/common';
+import { AppScreenName, AppColor, AppIcon } from '~/common/enums/enums';
+import { styles } from './styles';
+
+const CustomDrawerContent: FC<DrawerContentComponentProps> = (props) => {
+  const focusedRoute = props.state.routes[props.state.index];
+
+  return (
+    <ScrollView style={styles.container}>
+      <View style={styles.header}>
+        <Image source={AppIcon.LOGO} />
+      </View>
+      <View style={styles.list}>
+        <Text style={styles.listTitle}>Menu</Text>
+        <DrawerItem
+          style={styles.listItem}
+          activeTintColor={AppColor.TEXT.GRAY_100}
+          inactiveTintColor={AppColor.TEXT.GRAY_200}
+          activeBackgroundColor={AppColor.BRAND.BLUE_100}
+          focused={focusedRoute.name === AppScreenName.OVERVIEW ? true : false}
+          label={({ color }): JSX.Element => (
+            <Text style={{ color, ...styles.listItemText }}>
+              {AppScreenName.OVERVIEW}
+            </Text>
+          )}
+          icon={({ focused }): JSX.Element => (
+            <Image
+              style={styles.icon}
+              source={focused ? AppIcon.HOME_LIGHT : AppIcon.HOME_DARK}
+            />
+          )}
+          onPress={(): void =>
+            props.navigation.navigate(AppScreenName.OVERVIEW)
+          }
+        />
+        <DrawerItem
+          style={styles.listItem}
+          activeTintColor={AppColor.TEXT.GRAY_100}
+          inactiveTintColor={AppColor.TEXT.GRAY_200}
+          activeBackgroundColor={AppColor.BRAND.BLUE_100}
+          focused={focusedRoute.name === AppScreenName.COURSES ? true : false}
+          label={({ color }): JSX.Element => (
+            <Text style={{ color, ...styles.listItemText }}>
+              {AppScreenName.COURSES}
+            </Text>
+          )}
+          icon={({ focused }): JSX.Element => (
+            <Image
+              style={styles.icon}
+              source={focused ? AppIcon.BOOK_LIGHT : AppIcon.BOOK_DARK}
+            />
+          )}
+          onPress={(): void => props.navigation.navigate(AppScreenName.COURSES)}
```

Comment:
If i split function i get error ![image](https://i.gyazo.com/56ea895bea3337cf4e86791775894095.png)

Reply:
It's because you pass function result, not a function itself. You should move it outside. Also if you separate drawer item component, then you can pass screen name in props, and you will need only one handler
```
const handleItemPress = () => {
  navigate(screen)
}

//

onPress={handleItemPress}
```

Reply:
Seems like your handleNavigateTo looks like 
```
const handleNavigateTo = (name: AppScreenName): void => {
  props.navigation.navigate(name);
}
```
So you immediately call this handler instead of passing it,, so if you what to make a closure for handler on `AppScreenName` - it should look like this
```
const handleNavigateTo = (name: AppScreenName) => (): void => {
  props.navigation.navigate(name);
}
```